rsc_table/resource_table.c

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
I feel like these files would be better to live in the Zephyr code base and not here.  So maybe adding them in lib/open-amp/

Reply:
Thanks for the suggestion. I was not convince either that this was the best place for these files....
So I'll move them, and close this pull request